vars/buildKibanaDockerImage.groovy

```diff
@@ -0,0 +1,125 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+import org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper
+import hudson.model.Result
+import co.elastic.BuildException
+
+/**
+
+  Builds the Docker image for Kibana, from a branch or a pull Request.
+
+  buildKibanaDockerImage(branch: 'master')
+  buildKibanaDockerImage(branch: 'PR/12345')
+*/
+def call(Map args = [:]){
+  def branch = args?.branch?.trim() ? args.branch : 'master'
+  def uppercaseBranch = branch.toUpperCase()
+
+  def kibanaBranch = branch
+  if (uppercaseBranch.startsWith('PR/')) {
+    kibanaBranch = uppercaseBranch
+  }
+  log(level: 'INFO', text: "Kibana branch is: ${kibanaBranch}")
+
+  //buildKibana(branch: "${kibanaBranch}")
+}
+
+def buildKibana(Map args = [:]) {
+  def branch = args.branch
```

Comment:
maybe refspec instead branch? Similar to https://github.com/elastic/apm-pipeline-library/pull/1040/files#diff-80a4e01c27003c343412740cfda4b056168ff338c9010344c637a9b10ca94a07R75

Reply:
Addressed in 502af011f5110cfa4cb9d281f005aea75a85ec87